frontend/src/libs/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,68 @@
+import Select, { type StylesConfig } from 'react-select';
+
+import { useCallback, useState } from '~/libs/hooks/hooks.js';
+
+type Option = { label: string; value: string };
+
+type Properties = {
+  options: Option[];
+};
+
+const Dropdown: React.FC<Properties> = ({
+  options,
+}: Properties): JSX.Element => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const onMenuOpen = useCallback(() => setIsMenuOpen(true), []);
+  const onMenuClose = useCallback(() => setIsMenuOpen(false), []);
+
+  const stylesConfig: StylesConfig = {
```

Comment:
Move it out of FC and create function `getStyles({isMenuOpen}): StylesConfig` above the component.
Optimize function calls by using useCallback